singularity-eos/eos/modifiers/ramps_eos.hpp

```diff
@@ -428,12 +428,13 @@ class BilinearRampEOS : public EosBase<BilinearRampEOS<T>> {
   // Vector functions that overload the scalar versions declared here.
   SG_ADD_BASE_CLASS_USINGS(BilinearRampEOS<T>)
 
-  PORTABLE_FORCEINLINE_FUNCTION
-  bool IsModified() const { return true; }
-  PORTABLE_FORCEINLINE_FUNCTION
-  T UnmodifyOnce() { return t_; }
-  PORTABLE_FORCEINLINE_FUNCTION
-  auto GetUnmodifiedObject() { return t_.GetUnmodifiedObject(); }
+  inline constexpr bool IsModified() const { return true; }
+
+  inline constexpr T UnmodifyOnce() { return t_; }
+
+  inline constexpr decltype(auto) GetUnmodifiedObject() {
```

Comment:
`decltype(auto)` is something I haven't seen before. Interesting. Also didn't appreciate these can be `constexpr`. Why is that the case?